### PR TITLE
[ADZ-1240] API spec branding: request parameters

### PR DESCRIPTION
**Pull request by ziemsky**
Also fixes:
* [ADZ-1193] API spec branding: response headers
* [ADZ-1199] API spec: certain integer values are
  not displayed correctly

### CODE REVIEW

File: cms/src/main/resources/api-specification/codegen-templates/index.mustache [hunk 1]
```diff
@@ -41,74 +41,58 @@
                                             <p>{{{markdown notes levelToNormaliseHeadingsTo=4}}}</p>
 
                                             {{#isAnyTrue this hasPathParams hasHeaderParams hasBodyParam hasFormParams hasQueryParams}}
-                                                <h4>Request</h4>
+                                                <h4 class="nhsd-t-heading-l">Request</h4>
```
Comment (harryhmc) [inline cms/src/main/resources/api-specification/codegen-templates/index.mustache:44]: i believe the guidance is for h4's to use the class `nhsd-t-heading-m`?

File: cms/src/main/resources/api-specification/codegen-templates/params_table.mustache [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<h{{variable "paramsHeadingLevel"}} class="nhsd-t-heading-m">{{variable "paramsHeading"}}</h{{variable "paramsHeadingLevel"}}>
```
Comment (harryhmc) [inline cms/src/main/resources/api-specification/codegen-templates/params_table.mustache:1]: i see that this is hardcoded as a `nhsd-t-heading-m` heading also, is that intentional?

**Review by harryhmc — APPROVED** [review]

File: cms/src/main/resources/api-specification/codegen-templates/params_table.mustache (hunk 2 above)
Comment (ziemsky) [reply cms/src/main/resources/api-specification/codegen-templates/params_table.mustache:1]: Another good spot! :+1: 

No, it was not, I'm afraid.

I've now moved the whole heading line out of this template and into the caller one (`index.mustache`), applying correct levels and classes there. At first I've introduced yet another variable for the CSS classes to be set before invoking this template, but this increased the boilerplate code to total of three variables, all targeting just the heading line; seeing how excessive this seemed, I got rid of them all and just moved the line outside of the template.

File: cms/src/main/resources/api-specification/codegen-templates/index.mustache (hunk 1 above)
Comment (ziemsky) [reply cms/src/main/resources/api-specification/codegen-templates/index.mustache:44]: Indeed, good point!

Now fixed, together with a bunch of other `<hX>` elements in the template.

**Comment by ziemsky** [comment]
I've applied some changes following your good observations, @harryhmc. I've kept them in a separate commit 
`d25f6bf` for now, so that it's clearer to see what they are.

I've also rebased the branch onto latest 'main rebrand' one, so the commit you've reviewed is now showing as `597c49b`.

I'll squash them into one before re-integrating with the main feature branch.

**Comment by harryhmc** [comment]
thanks for the explanation of which commit to look at :)

all LGTM - you could probably write a helper to avoid hardcoding header classes but this would be more overhead than it's worth IMO

File: cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html [hunk 3]
```diff
@@ -20,185 +20,156 @@
...
         <h2 id="api-description__overview">Overview</h2>
-        <p>Use this API to access the <a href="https://digital.nhs.uk/services/demographics">Personal Demographics Service (PDS)</a></p>
-        <h2 id="api-description__legal-use">Legal use</h2>
-        <p>This API can only be used where there is a legal basis.
-    </div>
-
-    <div id="api-Patients" class="article-section">
-        <h2>Endpoints: Patients</h2>
+<p>Use this API to access the <a href="https://digital.nhs.uk/services/demographics">Personal Demographics Service (PDS)</a></p>
+<h2 id="api-description__legal-use">Legal use</h2>
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html:63]: Headers should use an appropriate header class, eg, `nhsd-t-heading-l`.

File: cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html [hunk 4]
```diff
@@ -207,193 +178,69 @@ <h6>Body</h6>
...
+                                                                <div class="body__schema__header">Schema
+                                                                    <span class="schema__header__buttons">
+                                                                        <button class='schemaButton js collapseAll' onclick="collapseAll('a7b833e4-f260-46f5-901e-2da15b11e5fd')">Collapse all</button>
+                                                                        <button class='schemaButton js expandAll' onclick="expandAll('a7b833e4-f260-46f5-901e-2da15b11e5fd')">Expand all</button>
                                                                     </span>
+                                                                </div>
+                                                                <div class="body__schema" data-schema-uuid="a7b833e4-f260-46f5-901e-2da15b11e5fd">
+                                                                    <table data-disablesort="true">
```
Comment (LEJA3) [inline cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html:188]: Tables should be wrapped in `nhsd-m-table`.

**Review by LEJA3 — COMMENTED** [review]
It looks like there are a few places where you should remove the reponsive classes from tables because you're using `data-responsive`. This would likely cause the issue you described where tables disappear at mobile viewports.

I also noticed a few places where headers, links and tables weren't being marked up correctly in the HTML document you sent over. Although, I'm not entirely sure where the issue is in the context of this PR.

- All headers should have a `nhsd-t-heading-*` token.
- All links should use the `nhsd-a-link` class.
- All tables should be wrapped in a `nhsd-m-table`.

File: cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html (hunk 4 above)
Comment (ziemsky) [reply cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html:188]: To be addressed in [ADZ-1194](https://nhsd-jira.digital.nhs.uk/browse/ADZ-1194).

File: cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html (hunk 3 above)
Comment (ziemsky) [reply cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html:63]: Addressed in [ADZ-1191](https://nhsd-jira.digital.nhs.uk/browse/ADZ-1191) / #1591.

File: cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html [hunk 5]
```diff
@@ -408,390 +255,205 @@ <h6>Body</h6>
...
+                                                <div class="method">get</div>
+                                                <div class="pre"><pre>/Patient</pre></div>
+                                                        <div class="try"><a onclick="tryEndpointNow('/Patients/search-patient')" class="ctabtn-left ctabtn--nhs-digital-button" onkeyup="return vjsu.onKeyUp(event)">Try this API</a></div>
+                                            </div>
+                                            <p><h4 id="endpoint">Endpoint</h4>
+<p>Use this endpoint to search for a patient in PDS.</p>
+                                                <h4 class="nhsd-t-heading-l">Request</h4>
+                                                <h5 class="nhsd-t-heading-m">Query parameters</h5>
+                                                <div class="nhsd-m-table nhsd-t-body">
+                                                    <table data-responsive data-no-sort class="nhsd-!t-display-hide nhsd-!t-display-s-show-table">
```
[the remark this answers is not in the document]
Comment (ziemsky) [reply cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html:360]: Will fix now :+1:

File: cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html [hunk 6]
```diff
@@ -408,390 +255,205 @@ <h6>Body</h6>
...
+                                                                <p class="nhsd-t-body">
+                                                                    Example: <span class="nhsd-a-text-highlight nhsd-a-text-highlight--code">true</span>
+                                                                </p>
+                                                            </td>
+                                                        </tr>
+                                                        <tr>
+                                                            <td><code>_history</code></td>
+                                                            <td>
+                                                                <p class="nhsd-t-body">
+                                                                    <strong>Boolean</strong>
```
[the remark this answers is not in the document]
Comment (ziemsky) [reply cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html:388]: Will fix now :+1:

[the remark this answers is not in the document]
Comment (ziemsky) [reply cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html]: Fixed.

File: cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (ziemsky) [reply cms/src/test/resources/test-data/api-specifications/ApiSpecSyncFromApigeeJobIntegrationTest/oasV3_complete.html]: Fixed.

**Pull request merged**